ibis/impala/udf.py

```diff
@@ -336,6 +341,31 @@ def _ibis_string_to_impala(tval):
         return repr(result)
 
 
+_COMPLEX_RE = re.compile(r'(struct|map|array)\<([\s\S]+)\>$')
+_STRUCT_FIELD_RE = re.compile(
+    r"`?(\w+)`?:((?:struct|map|array)\<[\s\S]+?\>|\S+)(?:\s+comment '([^']+)')?(?:,\n|$)") # noqa E501
+_MAP_FIELD_RE = re.compile(r"([^,]+),((?:struct|map|array)\<[\s\S]+?\>|.*)")
+
+
+def _impala_complex_type_to_ibis(tval):
+    t_type, t_schema = _COMPLEX_RE.findall(tval)[0]
+    if t_type == 'struct':
+        parts = _STRUCT_FIELD_RE.findall(t_schema)
+        names = [c[0] for c in parts]
+        types = [parse_type(c[1]) for c in parts]
+        t = dt.Struct(names, types)
+    elif t_type == 'map':
+        parts = _MAP_FIELD_RE.findall(t_schema)[0]
+        types = [parse_type(c) for c in parts]
+        t = dt.Map(types[0], types[1])
+    elif t_type == 'array':
+        type_ = parse_type(t_schema)
+        t = dt.Array(type_)
+    else:
+        raise Exception(tval)
```

Comment:
Can you handle errors in this function for each of the `t_type` cases?

Reply:
I raised errors for non matching regex, others should be raised by the corresponding call to the primitive type